### ISSUE TITLE
Can quantizer insert for call_function like torch.matmul or torch.nn.functional.softmax in model

### ISSUE THREAD

**Issue by bestzsq** [NVIDIA/TensorRT-Model-Optimizer#488]
### Insert TensorQuantizer for call_function like torch.matmul in model

I want to test quantization for some call_function like matmul or softmax, and float model was convert to quant model by iterating through model's named_children，however call_function can't found in model.named_children
```
def _replace_quant_module(model: nn.Module, version=None, registry=QuantModuleRegistry):
    """Helper function of replace_quant_module."""
    for name, child in model.named_children():
        if type(child) in registry:
            # REPLACE on the parent (model), not on child
            quantized = registry.convert(child)
            setattr(model, name, quantized)

        # now recurse into whichever module is now at `model.name`
        _replace_quant_module(getattr(model, name), version=version, registry=registry)
```
Here are some solutions I can think of：
**Method-1**：replace call_function like torch.matmul with a custom Matmul class that inherits from nn.Module，then matmul can be found in model.named_children()，and register QuantMatmul for Matmul in QuantModuleRegistry，so that it can be work well。

**Method-2**：replace call_function like torch.matmul with a custom Matmul class by fx.symbolic_trace，matmul can be found in  model.named_children(), simple model work well, however llm model like deepseek/qwen from huggingface do symbolic_trace failed as described in https://github.com/pytorch/pytorch/issues/149497

**Method-3**：register _QuantAttention for Qwen2Attention, so that kv cache(bmm) in attention can be quantized as described https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/4b522e0303d292cee973c83afb4e927a89fe6da6/modelopt/torch/quantization/plugins/huggingface.py#L52

These methods do not seem to be universally applicable，Is there any general method can do quantization for call_function？ 

Thank you in advance for your reply！

**Comment by realAsma** [comment]
> These methods do not seem to be universally applicable，Is there any general method can do quantization for call_function？

This is my thinking as well :) . 
Are you looking for a particular model? 

If you are asking specifically for `Qwen2Attention`? - I believe `Qwen2Attention` should be replaced by its quantized counter part. (also see https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/4b522e0303d292cee973c83afb4e927a89fe6da6/modelopt/torch/quantization/plugins/huggingface.py#L177)

**Comment by realAsma** [comment]
Your observations are correct - `ModelOpt` can only replace Pytorch `nn.Module` of a model. We do not do any graph tracing and insert quantization functions on the graph.

That said, if this operation (say torch.mamtul) happens a particular `nn.Module` such as say `T5Attention` , you can patch `torch.matmul` temporarily within `QuantT5Attention` forward function. This should be  the easiest method IMO.

For an example, please see the quantized implementation for `T5Attention` here - https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/4b522e0303d292cee973c83afb4e927a89fe6da6/modelopt/torch/quantization/plugins/huggingface.py#L143

I am closing this issue since it is answered. Please feel free to reopen.

**Comment by bestzsq** [comment]
@realAsma Thanks for your reply！Is there any future plans to support the quantization for call_function?
